test/storage/index_populator_test.cpp

```diff
@@ -0,0 +1,177 @@
+    // incompatible with return value of the BlockLayout::AttrSize
+    key_cols_0.emplace_back(IndexKeyColumn(col_oid, col.GetType(), false, INT8_MAX & col.GetAttrSize()));
+  }
+
+  {
+    auto col = table->GetSqlTable()->GetSchema().GetColumn("id");
+    auto col_oid = catalog::indexkeycol_oid_t(!col.GetOid());
+    key_cols_0.emplace_back(IndexKeyColumn(col_oid, col.GetType(), false));
+  }
+
+  {
+    auto col = table->GetSqlTable()->GetSchema().GetColumn("address");
+    auto col_oid = catalog::indexkeycol_oid_t(!col.GetOid());
+
+    // FIXME(xueyuan): Seems the Schema::GetAttrSize for varlen attribute is the internal representation, thus
+    // incompatible with return value of the BlockLayout::AttrSize
+    key_cols_0.emplace_back(IndexKeyColumn(col_oid, col.GetType(), false, INT8_MAX & col.GetAttrSize()));
+  }
+
+  {
+    auto col = table->GetSqlTable()->GetSchema().GetColumn("sex");
+    auto col_oid = catalog::indexkeycol_oid_t(!col.GetOid());
+    key_cols_0.emplace_back(IndexKeyColumn(col_oid, col.GetType(), false));
+  }
+
+  auto index_key_schema = IndexKeySchema(key_cols_0);
+
+  // Build the index
+  index_builder.SetOid(index_oid_0);
+  index_builder.SetConstraintType(ConstraintType::DEFAULT);
+  index_builder.SetKeySchema(index_key_schema);
+  auto index_0 = index_builder.Build();
+
+  // Populate the index
+  IndexPopulator::PopulateIndex(txn_, *table->GetSqlTable(), IndexKeySchema(key_cols_0), *index_0);
+
+  // Create the projected row for index
+  const IndexMetadata &metadata_0 = index_0->GetIndexMetadata();
+  const auto &pr_initializer = metadata_0.GetProjectedRowInitializer();
+  auto *key_buf_index = common::AllocationUtil::AllocateAligned(pr_initializer.ProjectedRowSize());
+  ProjectedRow *index_key = pr_initializer.InitializeRow(key_buf_index);
+
+  // Create the projected row for the sql table
+  std::vector<catalog::col_oid_t> col_oids;
+  col_oids.reserve(index_key_schema.size());
+  for (const auto &it : index_key_schema) {
+    col_oids.emplace_back(catalog::col_oid_t(!it.GetOid()));
+  }
+  auto init_and_map = table->GetSqlTable()->InitializerForProjectedRow(col_oids);
+  auto *key_buf = common::AllocationUtil::AllocateAligned(init_and_map.first.ProjectedRowSize());
+  ProjectedRow *key = init_and_map.first.InitializeRow(key_buf);
+
+  // Record the col_id of each column
+  std::vector<col_id_t> sql_table_cols;
+  sql_table_cols.reserve(key->NumColumns());
+  for (uint16_t i = 0; i < key->NumColumns(); ++i) {
+    sql_table_cols.emplace_back(key->ColumnIds()[i]);
+  }
+  for (const auto &it : *table->GetSqlTable()) {
+    if (table->GetSqlTable()->Select(txn_, it, key)) {
+      for (uint16_t i = 0; i < key->NumColumns(); ++i) {
+        key->ColumnIds()[i] = index_key->ColumnIds()[i];
+      }
+      std::vector<TupleSlot> tup_slots;
+      index_0->ScanKey(*key, &tup_slots);
+      bool flag = false;
+      for (const auto &tup_slot : tup_slots) {
+        if (it == tup_slot) {
```

Comment:
[Major] Is it better to also expect that there is only one tuple in returned `tup_slots`?

Reply:
No, since the index is not multi-versioned.